docs/sources/http_api/serviceaccount.md

```diff
@@ -0,0 +1,341 @@
+
+```http
+POST /api/serviceaccounts HTTP/1.1
+Accept: application/json
+Content-Type: application/json
+Authorization: Basic YWRtaW46YWRtaW4=
+
+{
+  "name": "grafana",
+  "role": "Admin",
+}
+```
+
+Requires basic authentication and that the authenticated user is a Grafana Admin.
+
+**Example Response**:
+
+```http
+HTTP/1.1 200
+Content-Type: application/json
+
+{
+	"id": 1,
+	"name": "test",
+	"login": "sa-test",
+	"orgId": 1,
+	"isDisabled": false,
+	"createdAt": "2022-03-21T14:35:33Z",
+	"updatedAt": "2022-03-21T14:35:33Z",
+	"avatarUrl": "/avatar/8ea890a677d6a223c591a1beea6ea9d2",
+	"role": "Viewer",
+	"teams": []
+}
+```
+
+## Get single serviceaccount by Id
+
+`GET /api/serviceaccounts/:id`
+
+#### Required permissions
+
+See note in the [introduction]({{< ref "#serviceaccount-api" >}}) for an explanation.
+
+| Action               | Scope              |
+| -------------------- | ------------------ |
+| serviceaccounts:read | serviceaccounts:\* |
+
+**Example Request**:
+
+```http
+GET /api/serviceaccounts/1 HTTP/1.1
+Accept: application/json
+Content-Type: application/json
+Authorization: Basic YWRtaW46YWRtaW4=
+```
+
+Requires basic authentication and that the authenticated user is a Grafana Admin.
+
+**Example Response**:
+
+```http
+HTTP/1.1 200
+Content-Type: application/json
+
+{
+	"id": 1,
+	"name": "test",
+	"login": "sa-test",
+	"orgId": 1,
+	"isDisabled": false,
+	"createdAt": "2022-03-21T14:35:33Z",
+	"updatedAt": "2022-03-21T14:35:33Z",
+	"avatarUrl": "/avatar/8ea890a677d6a223c591a1beea6ea9d2",
+	"role": "Viewer",
+	"teams": []
+}
+```
+
+## Update service account
+
+`PATCH /api/serviceaccounts/:id`
+
+#### Required permissions
+
+See note in the [introduction]({{< ref "#serviceaccount-api" >}}) for an explanation.
+
+| Action                | Scope              |
+| --------------------- | ------------------ |
+| serviceaccounts:write | serviceaccounts:\* |
+
+**Example Request**:
+
+```http
+PUT /api/serviceaccounts/2 HTTP/1.1
+Accept: application/json
+Content-Type: application/json
+Authorization: Basic YWRtaW46YWRtaW4=
+
+{
+  "name": "test",
+	"role": "Editor"
+}
+```
+
+Requires basic authentication and that the authenticated user is a Grafana Admin.
+
+**Example Response**:
+
+```http
+HTTP/1.1 200
+Content-Type: application/json
+
+{
+	"id": 2,
+	"name": "test",
+	"login": "sa-grafana",
+	"orgId": 1,
+	"isDisabled": false,
+	"createdAt": "2022-03-21T14:35:44Z",
+	"updatedAt": "2022-03-21T14:35:44Z",
+	"avatarUrl": "/avatar/8ea890a677d6a223c591a1beea6ea9d2",
+	"role": "Editor",
+	"teams": []
+}
+```
+
+---
+
+## Service account tokens
+
+## Get service account tokens
+
+`GET /api/serviceaccounts/:id/tokens`
+
+#### Required permissions
+
+See note in the [introduction]({{< ref "#serviceaccount-api" >}}) for an explanation.
+
+| Action               | Scope              |
+| -------------------- | ------------------ |
+| serviceaccounts:read | serviceaccounts:\* |
+| apikeys:read         | apikeys:\*         |
```

Comment:
I know this is not strictly related to this PR, however curious to know - is `apikeys:read` action and it's scope used also for the API Keys and we have reused it for service account tokens? I'm just wondering if having a new action for tokens would make sense, to keep API Keys separate for future maintainability.

Reply:
hmm 🤔 this is a good thought. It might make sense to have it as a separate scope. As we have for most things, folder/dashboards even though they are the same table.
We might want to refactor this actually.
Add a issue: https://github.com/grafana/grafana-enterprise/issues/3120

Reply:
Just added a comment in the issue, but it seems that the apikeys:read is not used at all for service accounts ?

Reply:
yea this is true. it is only "serviceaccounts:read/write" that is needed, which is more okay.

Reply:
will change the doc here